### ISSUE TITLE
Save states non-persistent after restart with all retroarch cores

### ISSUE THREAD

**Issue by Sorntor** [ClusterM/hakchi2#360]
Here is telnet output for various requested commands....

find /var/lib
mount
df -h

https://pastebin.ca/3809725

Save states persist so long as classic stays powered on. They disappear when power cycled.

**Comment by ClusterM** [comment]
Hi. 
Thanks for created issue, i will not forget about your problem that way.
Original NES games saves working without problems? Non-original NES-games with original emulator?

**Comment by Sorntor** [comment]
Correct, NES games using the built-in classic NES saving system all work properly. So if its a permissions issue, the permissions on that side is great. 

Is there any area that you would like me to write files to to see if they stay persistent after reboot? I figure some areas are nonvol that are wiped at boot, and others are jffs or some other writeable partition that stays persistent. Maybe I can test a little bit to see what sticks and what does not? Obviously the area where retroarch saves its save states is wiped at boot?

Also, update for you I've done some more testing. I've come to realize that the save states do not stick once you exit the game. As soon as I exit and reload the save states are gone. As long as I stay in-game then the save states stay persistent no matter which save slot 0-9 is used.

Also, regular game saves (I think those are .SRM?) DO stick! So I can power off the classic and power it back on and as long as the game has a built in save function then retroarch will correctly store the .srm and reload it at next launch.

It is only the save state function which is not working and this is across every retroarch core.

**Comment by elierodrigue** [comment]
Did you applied the save game cleanup fix from helping hand?

**Comment by Sorntor** [comment]
@elierodrigue - I've tried searching to better understand what you are referencing and could find nothing. Could you please point to me this fix?

**Comment by elierodrigue** [comment]
There's a strange free space leak with the way games are managed and I did a fix in the Helping Hand software : https://github.com/elierodrigue/Hakchi_HelpingHand
As the fix is clearing out saved games it might have been related but if you didnt knew about it then its not, and its not worth giving a try as the fix is not related to your issue.

**Comment by Sorntor** [comment]
Right, I did not know this existed so should be unrelated :) Thank you for your thoughts regardless.

**Comment by Sorntor** [comment]
@ClusterM  - Is there anything else I can provide on this issue? Happy to let you rdp to take a look if you want. I would get a new classic, but obviously these things are like rare precious metals at this point, im never getting another.

Thanks!

**Comment by war3zlod3r** [comment]
I had a similar issue recently, my suspend points would just launch the game without loading the point, I reset and reinstalled hackchi and restored the suspend points using the save state manager and that got it working again however if I use retroarch save states they work until I exit the game and then if I go back in are gone, any ideas why this would be happening or how to fix?

I tried the helping hand fix and it didn't work for me note this is on a SNES mini

**Comment by cryptolute** [comment]
After researching I determined the non persistent saves via retroarch is
normal behavior and there is no fix. You can only use suspend points with
exception to n64 I think it was? That only allows sram saves (in game) I
believe.....

**Comment by war3zlod3r** [comment]
I don't think this is the case I reinstalled the latest retroarch cores and now all is working correctly minus fast forward

**Comment by cryptolute** [comment]
You are saying your retroarch save states stay after reboot? Which cores
are you using? Would love to confirm.

**Comment by war3zlod3r** [comment]
Yes they do!

**Comment by cryptolute** [comment]
Please provide to me what retroarch core you are using and what emulator
cores you are testing with. I need version, source, etc. I would like to
test and confirm myself thank you.

**Comment by war3zlod3r** [comment]
The pack downloaded with retroarch-clover SNES9x 2010

**Comment by cryptolute** [comment]
That is not a good answer. Provide a URL where you obtained it. I have no
idea what version of "retroarch-clover" you are referring to, or where you
obtained it.

**Comment by war3zlod3r** [comment]
The latest version from Retroarch-clovers github, I'm using 1.0c ... the latest, if my answers aren't good though I'll keep them to myself

**Comment by cryptolute** [comment]
Can you try to do a save state in n64 and tell us your results?

Thanks

**Comment by Drakonas** [comment]
@cryptolute he's talking about Cluster's retroarch-clover official coreset here on Github and linked on the Hakchi releases page: https://github.com/ClusterM
